src/server/game/Entities/Player/Player.cpp

```diff
@@ -10463,28 +10463,11 @@ uint8 Player::FindEquipSlot(ItemTemplate const* proto, uint32 slot, bool swap) c
             break;
         case INVTYPE_2HWEAPON:
             slots[0] = EQUIPMENT_SLOT_MAINHAND;
-            if (Item* mhWeapon = GetItemByPos(INVENTORY_SLOT_BAG_0, EQUIPMENT_SLOT_MAINHAND))
-            {
-                if (ItemTemplate const* mhWeaponProto = mhWeapon->GetTemplate())
-                {
-                    if (mhWeaponProto->SubClass == ITEM_SUBCLASS_WEAPON_POLEARM || mhWeaponProto->SubClass == ITEM_SUBCLASS_WEAPON_STAFF)
-                    {
-                        const_cast<Player*>(this)->AutoUnequipOffhandIfNeed(true);
```

Comment:
Can you still equip it okay if you have spots in your bags and an offhand equipped as you remove AutoUnequipOffhandIfNeed?

Reply:
I tested it. (Druid with 1H-Mace/Offhand -> Staff, Warrior with 1H-Sword/Shield -> 2H Axe/Polearm)
`AutoUnequipOffhandIfNeed()` gets called by the functions which call `FindEquiptSlot()`.

`FindEquipSlot()` is called in `CanEquipItem()` which gets called in either `SwapItem()` (Player.cpp) or `HandleAutoEquipItemOpcode()` (ItemHandler.cpp). Both functions also call `AutoUnequipOffhandIfNeed()` on their own.